browser/importer/brave_external_process_importer_host.cc

```diff
@@ -5,178 +5,65 @@
-          base::BindOnce(
-              &BraveExternalProcessImporterHost::OnExtensionInstalled,
-              weak_ptr_factory_.GetWeakPtr(), id));
-  installer->BeginInstall();
-}
-
-void BraveExternalProcessImporterHost::ImportExtensions(
-    std::vector<std::string> ids) {
-  for (const auto& id : ids) {
-    InstallExtension(id);
-  }
-
-  if (!ids.empty() && observer_)
-    observer_->ImportItemEnded(importer::EXTENSIONS);
-
-  ExternalProcessImporterHost::NotifyImportEnded();
-}
-
-void BraveExternalProcessImporterHost::OnGetChromeExtensionsList(
-    std::optional<std::vector<std::string>> extensions_list) {
-  if (!extensions_list.has_value()) {
-    ExternalProcessImporterHost::NotifyImportEnded();
-    return;
-  }
-  const auto ids = extensions_list.value();
-  if (ids.empty()) {
-    ExternalProcessImporterHost::NotifyImportEnded();
-    return;
-  }
-  extensions::GetExtensionFileTaskRunner()->PostTaskAndReply(
-      FROM_HERE,
-      base::BindOnce(&brave::ImportStorages, source_profile_.source_path,
-                     profile_->GetPath(), ids),
-      base::BindOnce(&BraveExternalProcessImporterHost::ImportExtensions,
-                     weak_ptr_factory_.GetWeakPtr(), ids));
-}
-
 void BraveExternalProcessImporterHost::NotifyImportEnded() {
-  if (!cancelled_)
+  if (!cancelled_) {
     RecordImporterP3A(source_profile_.importer_type);
+  }
 
   // If user chooses extension importing, start importing extensions.
   // and NotifyImportEnded() will be called from OnGetChromeExtensionsList().
   // Handling extensions importing after finishing all other properties makes
   // logic simpler.
   // Don't import if cancelled.
-  if (!cancelled_ && (items_ & importer::EXTENSIONS)) {
-    LaunchExtensionsImport();
-    return;
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+  if (NeedToImportExtensions() && extensions_importer_) {
+    NotifyImportItemStarted(importer::EXTENSIONS);
+    if (extensions_importer_->Import(base::BindRepeating(
+            &BraveExternalProcessImporterHost::OnExtensionImported,
+            weak_ptr_factory_.GetWeakPtr()))) {
+      return;
+    }
   }
+#endif
 
   // Otherwise, notifying here and importing is finished.
   ExternalProcessImporterHost::NotifyImportEnded();
 }
 
 void BraveExternalProcessImporterHost::LaunchImportIfReady() {
-  if (do_not_launch_import_for_testing_)
-    return;
-  ExternalProcessImporterHost::LaunchImportIfReady();
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+  if (NeedToImportExtensions()) {
+    if (!extensions_importer_) {
+      extensions_importer_ =
+          std::make_unique<extensions_import::ExtensionsImporter>(
+              source_profile_.source_path, profile_);
+      return extensions_importer_->Prepare(base::BindOnce(
+          &BraveExternalProcessImporterHost::OnExtensionsImportReady,
+          weak_ptr_factory_.GetWeakPtr()));
+    }
+  }
+#endif
+
+  if (!do_not_launch_import_for_testing_) {
+    ExternalProcessImporterHost::LaunchImportIfReady();
```

Comment:
Not clear what happens here. It's **External**ProcessImporterHost, so it (re)creates `BraveExternalProcessImporterClient`?
 
 If so, we should be careful with retries (when we call `LaunchImportIfReady()`).

Reply:
Yes it recreates BraveExternalProcessImporterClient